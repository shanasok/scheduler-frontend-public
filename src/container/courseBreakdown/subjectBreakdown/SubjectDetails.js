import React from 'react';
import SubjectName from './SubjectName';
import SubjectSummary from './SubjectSummary';
import LessonRibbon from './LessonRibbon';

class SubjectDetails extends React.Component {

    render() {
        return <div className="subject">
            <SubjectName name={this.props.subject.name} />
            <SubjectSummary summary={this.props.subject.summary} />
            <LessonRibbon lessons={this.props.subject.lessons} />
        </div>
    }
}

export default SubjectDetails;