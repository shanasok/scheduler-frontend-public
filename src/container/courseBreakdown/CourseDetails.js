import React from 'react';
import { withRouter } from "react-router-dom";
import SubjectDetails from "./subjectBreakdown/SubjectDetails";
import CourseName from "./CourseName";
import CourseSummary from "./CourseSummary";
import '../../css/CourseDetailsStyles.css';

export function sortLowestNumberFirst(subjects) {
    return subjects.sort(function(a, b){return a.subjectNumber - b.subjectNumber});
}
//.sort(function(a, b){return a.subjectNumber - b.subjectNumber}) //lowest number first
export function populateCourseSubjects(subjects, sortFunction){
    if (subjects && Array.isArray(subjects)) {
        sortFunction(subjects);
        return subjects
            .map(subject =>
                <SubjectDetails key={'subject'+subject.id} subject={subject}/>
            )
    }
}

class CourseDetails extends React.Component {

    render() {
        const courseName = this.props.location.state.course.name;
        const courseSummary = this.props.location.state.course.summary;
        const subjects = this.props.location.state.course.subjects;
        const courseSubjects = populateCourseSubjects(subjects, sortLowestNumberFirst);

        return <div className="courseDetails">
            <CourseName courseName={courseName} />
            <CourseSummary courseSummary={courseSummary} />
            { courseSubjects }
        </div>
    }
}

export default withRouter(CourseDetails);