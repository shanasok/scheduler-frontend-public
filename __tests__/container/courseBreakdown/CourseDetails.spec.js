import React from 'react';
import CourseDetails, {populateCourseSubjects, sortLowestNumberFirst} from '../../../src/container/courseBreakdown/CourseDetails';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from "history";
import SubjectDetails from "../../../src/container/courseBreakdown/subjectBreakdown/SubjectDetails";


it('renders correctly', () => {

    const history = createMemoryHistory();
    history.location.state = {
        course: {
            name: "testCourseName",
            summary: "testCourseSummary",
            subjects: [
                {
                    name: "testSubjectName",
                    summary: "testSubjectSummary",
                    subjectNumber: 1
                }
            ]
        }
    }
    const tree = renderer
        .create(<Router history={history}><CourseDetails /></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('populateCourseSubjects', () => {

    const testSubjects = [
        {
            name: "testSubjectName",
            summary: "testSubjectSummary",
            subjectNumber: 1
        }
    ];

    it('should return undefined if subjects is empty', () => {
        const subjects = null;
        expect(populateCourseSubjects(subjects)).toBeUndefined();
    });

    it('should return map of size 1 when passing in 1 course', () => {
        const result = populateCourseSubjects(testSubjects, sortLowestNumberFirst);
        expect(result.length).toBe(1);
    })

    it('should call sort one time', () => {
        const mockSortFunction = jest.fn(sortLowestNumberFirst);
        const result = populateCourseSubjects(testSubjects, mockSortFunction);
        expect(mockSortFunction.mock.calls.length).toBe(1);
    })
})