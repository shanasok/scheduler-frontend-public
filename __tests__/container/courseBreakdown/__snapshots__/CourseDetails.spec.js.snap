// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="courseDetails"
>
  <div
    className="courseName"
  >
    testCourseName
  </div>
  <div
    className="courseSummary"
  >
    testCourseSummary
  </div>
  <div
    className="subject"
  >
    <div
      className="subjectName"
    >
      testSubjectName
    </div>
    <div
      className="subjectSummary"
    >
      testSubjectSummary
    </div>
    <div
      className="lessonRibbon"
    />
  </div>
</div>
`;
